### ISSUE TITLE
Block Keys Reset On /is admin recalc

### ISSUE THREAD

**Issue by Justugh** [BG-Software-LLC/SuperiorSkyblock2#1022]
### Minecraft's Version

Paper version git-Paper-267

### Plugin's Version

SuperiorSkyblock2 1.9.0

### Describe the bug

Creating a custom block key works, however on /is admin recalc the block key is gone.

### To Reproduce

Create a custom block key, register it and then run /is admin recalc.

### Additional Information

[CustomKeyParser](https://pastebin.com/qHxMLthr)
[KeyParser Registration](https://pastebin.com/QAwLrnAU)

**Comment by OmerBenGera** [comment]
Mind uploading your custom module/plugin to mediafire or something so I can test it out?

**Comment by Justugh** [comment]
Well it's modified now to get around the issue, however I can upload what I have.  The concept is still the same, however we don't use the block values to add value, instead we use bonus worth to get around the issue I listed above.

Project is here: [GitHub Repo](https://github.com/Justugh/CustomSBBlocks)

**Comment by OmerBenGera** [comment]
> Well it's modified now to get around the issue, however I can upload what I have. The concept is still the same, however we don't use the block values to add value, instead we use bonus worth to get around the issue I listed above.
> 
> Project is here: [GitHub Repo](https://github.com/Justugh/CustomSBBlocks)

Awesome.
Mind explaining how to get the "custom block" to be saved and counted as custom one?
What do I need to do in order to replicate the issue using your plugin?

**Comment by Justugh** [comment]
So essentially the config should come with a generic "Example-TNT". The plugin requires a valid SQL connection to store the block data.

- Add Example-TNT to your block-values.yml, then reload skyblock.
- You can then use /cstmb give <username> Example-TNT.
- Place it, check your island worth, I also used /is blocks to make sure it was actually being recognized.
- Then use /is recalc and it'll disappear from your worth & from /is blocks. It'll appear as normal TNT.

Also, I tried Example_TNT as well, as the key to make sure it wasn't some syntax issue and the issue persisted.

**Comment by OmerBenGera** [comment]
> So essentially the config should come with a generic "Example-TNT". The plugin requires a valid SQL connection to store the block data.
> 
>     * Add Example-TNT to your block-values.yml, then reload skyblock.
> 
>     * You can then use /cstmb give <username> Example-TNT.
> 
>     * Place it, check your island worth, I also used /is blocks to make sure it was actually being recognized.
> 
>     * Then use /is recalc and it'll disappear from your worth & from /is blocks. It'll appear as normal TNT.
> 
> 
> Also, I tried Example_TNT as well, as the key to make sure it wasn't some syntax issue and the issue persisted.

I can't compile your project as `JustAPI-Spigot` is missing from the repositories.
Is there a repository that it is hosted on?

**Comment by OmerBenGera** [comment]
I tried to add the jar as a local dependency and I receive errors about a missing symbol in SafeLocation - therefore it doesn't seem like it's really possible atm to build the repository myself. Mind sending me a compiled jar?

**Comment by Justugh** [comment]
Apologies, I'll go ahead and send my current jar.
[CustomSBBlocks-2022.3.zip](https://github.com/BG-Software-LLC/SuperiorSkyblock2/files/8401223/CustomSBBlocks-2022.3.zip)

**Comment by OmerBenGera** [comment]
The issue was that you register your key-parser with keys of items, which add "0" as a subkey, but SSB deducts the block data when counting blocks. I advice not to use keys of items and instead get the global key of these keys and register your key parser against them. However, this is still an issue and therefore I pushed an update to SSB to not ignore the block data when counting blocks, and set it to 0.

In conclusion, fixed in latest dev build :)
